test/integration/hooks/associations.test.js

```diff
@@ -62,14 +63,14 @@ describe(Support.getTestDialectTeaser('Hooks'), function() {
           , beforeHook = false
           , afterHook = false;
 
-          this.Tasks.beforeUpdate(function(task, options, fn) {
+          this.Tasks.beforeUpdate(function(task, options) {
             beforeHook = true;
-            fn();
+            return Promise.resolve();
```

Comment:
Recursion is not relevant here? We wrap whatever response in Promise.resolve() so if it returns undefined it's just an undefined result in the resolved promise.